sandbox/modules/identity/build.gradle

```diff
@@ -11,13 +11,14 @@ apply plugin: 'opensearch.internal-cluster-test'
 opensearchplugin {
   description 'Plugin for identity features in OpenSearch.'
   classname 'org.opensearch.identity.IdentityPlugin'
-  name project.name
+  name 'identity'
   licenseFile rootProject.file('licenses/APACHE-LICENSE-2.0.txt')
   noticeFile rootProject.file('NOTICE.txt')
+  extendedPlugins = ['extensible-identity']
 }
 
 dependencies {
-  implementation project(':sandbox:libs:opensearch-authn')
+  compileOnly project(':sandbox:libs:opensearch-authn')
```

Comment:
In this PR,  the `identity` module extends the `extensible-identity` module which has this as an `implementation` dependency. It will get the dependency at runtime because of this this relationship.